packages/envision/src/js/dropdown.js

```diff
@@ -3,240 +3,165 @@
  * Licensed under MIT https://github.com/sitevision/envision/blob/master/LICENSE
  * --------------------------------------------------------------------------
  */
+import { getNodes, hide, unhide } from './util/nodes';
 
-import $ from 'jquery';
-
-const NAME = 'envDropdown';
 const ENV_DROPDOWN_OPEN = 'env-is-open';
 const TOGGLE_DROPDOWN = '[data-dropdown]';
-const ENV_DROPDOWN = '.env-dropdown';
-const ENV_CLICK_EVENT = 'click.env-dropdown';
-const ENV_KEYDOWN_EVENT = 'keydown.env-dropdown';
-const ENV_KEYUP_EVENT = 'keyup.env-dropdown';
+const TARGET_ATTR = 'data-target';
+const INDEX_ATTR = 'data-env-dropdown-index';
 const ENV_DROPDOWN_MENU = '.env-dropdown__menu';
-const RIGHT_MOUSE_BUTTON = 2;
-const KEYS = {
-   ESCAPE_KEY: 'Escape',
-   ARROWDOWN: 'ArrowDown',
-   ARROWUP: 'ArrowUp',
-   SPACE: 32,
-   TAB_KEY: 'Tab',
-};
-const REGEXP_KEYDOWN = new RegExp(
-   `${KEYS.ARROWUP}|${KEYS.ARROWDOWN}|${KEYS.ESCAPE_KEY}`
-);
+const DATA_INITIALIZED = 'data-env-dropdown';
 
-class Dropdown {
-   constructor(element) {
-      this.el = element;
-      this.$el = $(element);
-      this._id = this.el.id;
-   }
+const dropdowns = {};
 
-   toggle() {
-      if (this.$el.hasClass(ENV_DROPDOWN_OPEN)) {
-         this.hide();
-      } else {
-         this.show();
-      }
+let Popper;
+const getPopper = async () => {
+   if (Popper) {
+      return Popper;
    }
 
-   show() {
-      this.$el.addClass(ENV_DROPDOWN_OPEN);
-
-      this.$el.find(TOGGLE_DROPDOWN).attr('aria-expanded', 'true');
-
-      this._bindEvents();
-   }
-
-   hide() {
-      this.$el.removeClass(ENV_DROPDOWN_OPEN);
+   const { default: DynamicPopper } = await import(
+      /* webpackChunkName: "popper" */ 'popper.js'
+   );
+   Popper = DynamicPopper;
+   return Popper;
+};
 
-      this.$el.find(TOGGLE_DROPDOWN).attr('aria-expanded', 'false');
+class Dropdown {
+   constructor(container, menu, button) {
+      this.button = button;
+      this.container = container;
+      this.placeholder = document.createElement('span');
+      this.menu = menu;
+      this.menuitems = getNodes('.env-dropdown__item', this.menu);
+      this.menu.insertAdjacentElement('beforebegin', this.placeholder);
+
+      this.menuitems.forEach((el, i) => {
+         el.setAttribute(INDEX_ATTR, i);
+      });
 
-      this._unbindEvents();
-   }
+      this._handleClick = (() => {
+         this.hide();
+      }).bind(this);
 
-   _bindEvents() {
-      const ENV_CLICK = `${ENV_CLICK_EVENT}-${this.el.id}`;
-      const ENV_KEYDOWN = `${ENV_KEYDOWN_EVENT}-${this.el.id}`;
+      this._handleKeyDown = ((e) => {
+         const el = e.target;
```

Comment:
_Åsikt_ Tycker att det är snyggare att lägga själva metoden någon annanstans och sen bara spara `this._handleKeyDown = keyHandler.bind(this)` i `constructor`. Det blir lättare att se vart konstruktorn börjar och slutar.